driver-core/src/main/java/com/datastax/driver/core/policies/TokenAwarePolicy.java

```diff
@@ -23,31 +23,70 @@
 import java.util.*;
 
 /**
- * A wrapper load balancing policy that add token awareness to a child policy.
+ * A wrapper load balancing policy that adds token awareness to a child policy.
  * <p/>
  * This policy encapsulates another policy. The resulting policy works in
  * the following way:
  * <ul>
  * <li>the {@code distance} method is inherited from the child policy.</li>
- * <li>the iterator return by the {@code newQueryPlan} method will first
- * return the {@code LOCAL} replicas for the query (based on {@link Statement#getRoutingKey})
- * <i>if possible</i> (i.e. if the query {@code getRoutingKey} method
- * doesn't return {@code null} and if {@link Metadata#getReplicas}
- * returns a non empty set of replicas for that partition key). If no
- * local replica can be either found or successfully contacted, the rest
- * of the query plan will fallback to one of the child policy.</li>
+ * <li>the iterator returned by the {@code newQueryPlan} method will first
+ * return the {@link HostDistance#LOCAL LOCAL} replicas for the query
+ * <em>if possible</em> (i.e. if the query's
+ * {@link Statement#getRoutingKey(ProtocolVersion, CodecRegistry) routing key}
+ * is not {@code null} and if the
+ * {@link Metadata#getReplicas(String, ByteBuffer) set of replicas}
+ * for that partition key is not empty). If no local replica can be either found
+ * or successfully contacted, the rest of the query plan will fallback
+ * to the child policy's one.</li>
  * </ul>
+ * The exact order in which local replicas are returned is dictated by the
+ * {@link ReplicaOrdering strategy} provided at instantiation.
  * <p/>
- * Do note that only replica for which the child policy {@code distance}
- * method returns {@code HostDistance.LOCAL} will be considered having
+ * Do note that only replicas for which the child policy's
+ * {@link LoadBalancingPolicy#distance(Host) distance}
+ * method returns {@link HostDistance#LOCAL LOCAL} will be considered having
  * priority. For example, if you wrap {@link DCAwareRoundRobinPolicy} with this
  * token aware policy, replicas from remote data centers may only be
- * returned after all the host of the local data center.
+ * returned after all the hosts of the local data center.
  */
 public class TokenAwarePolicy implements ChainableLoadBalancingPolicy {
 
+    /**
+     * Strategies for replica ordering.
+     */
+    public enum ReplicaOrdering {
+
+        /**
+         * Order replicas by token ring topology, i.e. always return the "primary" replica first,
+         * then the second, etc., according to the placement of replicas around the token ring.
+         * <p/>
+         * This strategy is the only one guaranteed to order replicas in a deterministic and
+         * constant way. This increases the effectiveness of server-side row caching (especially
+         * at consistency level ONE), but may create hotspots, since the primary replica is always
```

Comment:
nit: Would change `but may create hotspots` to `but is more heavily impacted by hotspots`.